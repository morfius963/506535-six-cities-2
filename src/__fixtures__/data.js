export default {
  cardItems: [
    {
      title: `Beautiful &amp; luxurious apartment at great location`
    },
    {
      title: `Wood and stone place`
    },
    {
      title: `Canal View Prinsengracht`
    },
    {
      title: `Nice, cozy, warm big bed apartment`
    },
  ],
  onTitleClick: () => {}
};
