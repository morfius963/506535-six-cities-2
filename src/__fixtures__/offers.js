const offers = [
  {
    city: `Amsterdam`,
    location: [52.38333, 4.9],
    offersCoords: [
      [52.3909553943508, 4.85309666406198],
      [52.369553943508, 4.85309666406198],
      [52.3909553943508, 4.929309666406198],
      [52.3809553943508, 4.939309666406198]
    ]
  },
  {
    city: `Paris`,
    location: [48.864716, 2.349014],
    offersCoords: [
      [48.854716, 2.339014],
      [48.874716, 2.399014],
      [48.844716, 2.319014]
    ]
  },
  {
    city: `Brussels`,
    location: [50.85045, 4.34878],
    offersCoords: [
      [50.81045, 4.30878],
      [50.83045, 4.39878],
      [50.87045, 4.32878],
    ]
  },
  {
    city: `Hamburg`,
    location: [53.551086, 9.993682],
    offersCoords: [
      [53.541086, 9.983682],
      [53.531086, 9.973682],
      [53.521086, 9.963682],
      [53.561086, 9.953682],
      [53.571086, 9.943682],
    ]
  },
  {
    city: `Dusseldorf`,
    location: [51.233334, 6.783333],
    offersCoords: [
      [51.243334, 6.773333],
      [51.253334, 6.763333],
      [51.263334, 6.753333],
      [51.273334, 6.743333]
    ]
  }
];

const makeMockOffer = (city, cityCoords, offersCoords, i) => ({
  id: i,
  city: {
    name: city,
    location: {
      latitude: cityCoords[0],
      longitude: cityCoords[1],
      zoom: 10
    }
  },
  title: `Beautiful &amp; luxurious apartment at great location`,
  isPremium: true,
  price: i * 1000 + 100,
  rating: i / 2,
  type: `apartment`,
  previewImage: `img/apartment-02.jpg`,
  images: [`img/apartment-01.jpg`, `img/apartment-02.jpg`],
  isFavorite: false,
  bedrooms: 1,
  maxAdults: 3,
  goods: [`1`, `2`, `3`],
  description: `Beautiful &amp; luxurious apartment at great location`,
  host: {
    id: i,
    isPro: true,
    name: `Me`,
    avatar: `img/apartment-02.jpg`
  },
  location: {
    latitude: offersCoords[0],
    longitude: offersCoords[1],
    zoom: 10
  }
});

export const sortValues = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];

export default offers.reduce((acc, {city, location, offersCoords}) => {
  const newOffers = offersCoords.map((coord, i) => makeMockOffer(city, location, coord, i));

  return [...acc, ...newOffers];
}, []);
