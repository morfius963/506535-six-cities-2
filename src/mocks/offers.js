const makeMockOffer = () => ({
  title: [`Beautiful &amp; luxurious apartment at great location`, `Wood and stone place`, `Canal View Prinsengracht`, `Nice, cozy, warm big bed apartment`][Math.floor(Math.random() * 4)],
  isPremium: Boolean(Math.round(Math.random())),
  price: 250,
  rating: Number((Math.random() * 5).toFixed(1)),
  type: [`apartment`, `private room`][Math.floor(Math.random() * 2)],
  previewImage: `img/apartment-0${Math.ceil(Math.random() * 3)}.jpg`
});

export default new Array(4).fill(``).map(makeMockOffer);
