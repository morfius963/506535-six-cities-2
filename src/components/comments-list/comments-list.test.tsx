import * as React from "react";
import * as renderer from "react-test-renderer";
import CommentsList from "./comments-list";
import CommentItem from "../comment-item/comment-item";
import {commentsData} from "../../__fixtures__/offers";

jest.mock(`../comment-item/comment-item`, () => jest.fn().mockReturnValue(null));

describe(`snapshot test`, () => {
  it(`Component correctly renders`, () => {
    const props = {
      comments: commentsData,
      id: 1,
      onCommentsLoad: jest.fn()
    };

    const tree = renderer
      .create(<CommentsList {...props} />)
      .toJSON();

    expect(CommentItem).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
