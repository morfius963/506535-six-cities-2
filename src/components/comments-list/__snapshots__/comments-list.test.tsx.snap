// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<ul
  className="reviews__list"
/>
`;
